src/service_daemon.rs

```diff
@@ -2356,6 +2356,15 @@ impl DnsCache {
             _ => return None,
         };
 
+        if incoming.get_cache_flush() {
+            // Mark all existing records of this type as expired, and prepend the new record.
+            record_vec.iter_mut().for_each(|r| {
+                r.expire();
```

Comment:
Per RFC 6762,  "...all old records with that name, rrtype,
   and rrclass that were received more than one second ago are declared
   invalid, and marked to expire from the cache in one second."

we should check for the same rrclass here,  and check for "received more than one second ago"  (You can use `DnsRecord.created` timestamp to check) .

Reply:
Checking when it was created, added `get_created()`